WordPress/WordPressTest/BlogServiceRemoteXMLRPCTest.swift

```diff
@@ -0,0 +1,271 @@
+
+        var isMultiAuthor = false
+        let response = [["ID" : 2], ["ID" : 3]]
+        blogServiceRemote.checkMultiAuthorWithSuccess( {
+                isMultiAuthor = $0
+            }, failure: nil)
+        mockApi.successBlockPassedIn?(response, NSHTTPURLResponse())
+        XCTAssertTrue(isMultiAuthor)
+    }
+
+    func testIsNotMultiAuthor() {
+
+        var isMultiAuthor = true
+        let response = [["ID" : 2]]
+        blogServiceRemote.checkMultiAuthorWithSuccess( {
+            isMultiAuthor = $0
+        }, failure: nil)
+        mockApi.successBlockPassedIn?(response, NSHTTPURLResponse())
+        XCTAssertFalse(isMultiAuthor)
+    }
+
+    func testSyncPostTypesRequest() {
+
+        blogServiceRemote.syncPostTypesWithSuccess(nil, failure: nil)
+        XCTAssertEqual(mockApi.methodPassedIn, "wp.getPostTypes")
+        XCTAssertTrue(checkParametersArray(mockApi.parametersPassedIn, expected: nil),
+                      "Default parameters not found in the parameter list")
+    }
+
+    func testSingleSyncPostTypes() {
+
+        let name = "name"
+        let response = [1 : [name : name]]
+        var remotePost = [RemotePostType]()
+        blogServiceRemote.syncPostTypesWithSuccess({
+            remotePost = $0
+        }, failure: nil)
+        mockApi.successBlockPassedIn?(response, NSHTTPURLResponse())
+        XCTAssertEqual(remotePost.count, 1)
+        XCTAssertEqual(remotePost[0].name, name)
+    }
+
+    func testMultipleSyncPostTypes() {
+
+        let name = "name"
+        let response = ["1" : [name : name], "2" : [ name : name]]
+        var remotePost = [RemotePostType]()
+        blogServiceRemote.syncPostTypesWithSuccess({
+            remotePost = $0
+        }, failure: nil)
+        mockApi.successBlockPassedIn?(response, NSHTTPURLResponse())
+        XCTAssertEqual(remotePost.count, 2)
+        XCTAssertEqual(remotePost[0].name, name)
+        XCTAssertEqual(remotePost[1].name, name)
+    }
+
+    func testSyncPostTypesFailure() {
+
+        let response = [:]
+        var fail = false
+        blogServiceRemote.syncPostTypesWithSuccess(nil, failure: { _ in
+            fail = true
+        })
+        mockApi.successBlockPassedIn?(response, NSHTTPURLResponse())
+        XCTAssertTrue(fail)
+    }
+
+    func testSyncPostFormatsRequest() {
+
+        let expectedParams = ["show-supported" : "1"]
+        blogServiceRemote.syncPostFormatsWithSuccess(nil, failure: nil)
+        XCTAssertEqual(mockApi.methodPassedIn, "wp.getPostFormats")
+        XCTAssertTrue(checkParametersArray(mockApi.parametersPassedIn, expected: expectedParams),
+                      "\(expectedParams) not found in the parameter list")
+    }
+
+    func testSyncPostFormatsAsArray() {
+
+        let response = ["supported" : ["f1", "f2", "f3"],
+                        "all" : ["f1" : "a" , "f2" : "b", "f3" : "c", "f4" : "d", "standard" : "e"]]
+        var responseFormats = [NSObject : NSObject]()
```

Comment:
I'm fine with having to cast the results, and it's even more explicit in stating what we expect to be in there.